### PR TITLE
[OTA] Integrate ota-requestor cluster as normal cluster in data model build flow

### PR DESCRIPTION
**Pull request by tima-q**
#### Problem
Building the ota-requestor cluster is currently done by directly including the sources from src/app/clusters at application level.
This generates code duplication in the .gn/cmake files when trying to add the requestor to another application.
Also ota-requestor-stubs with the same functionality as the general cluster exist.

#### Change overview
Introducing ota-requestor server in gn and cmake builds same way as other cluster are handled.
* Added to the list parsed in zap_cluster_list.py
* Adjusted chip_data_model.gni to select non-default sources of ota-requestor-server
* Adjust all-clusters-app and ota-requestor-app 
* changes to all-clusters similar to #13246, enabling provider-client to be able to use requestor-server. Opposed to that PR, keeping the ota-provider-server intact in this PR.

* Adjust tests to cope with ota-provider client that is added to all-clusters-app

This is preparation to add a ota-requestor-server to the QPG lighting-app. 
This PR holds the generic changes.

Note - CMake based builds of all-clusters-app are not using the default mechanism to include clusters. As it could be a conscious choice, leaving maintainers to update that further.

#### Testing
* CI builds + test runs
* Linux ota-provider <> ota-requestor-app functionality
** AnnounceOTAProvider triggered with chip-tool, processed at requestor side, OTAQuery and subsequent download seen.

### CODE REVIEW

**Review by selissia — APPROVED** [review]

**Comment by tima-q** [comment]
Keeping PR alive - need reviewers after holiday period

**Comment by tima-q** [comment]
Update to all-clusters part required due to issues in ZAP (client/cluster can't coexist) - see #13437 #13246.
@bzbarsky-apple input how to tackle this greatly appreciated

File: examples/all-clusters-app/esp32/main/CMakeLists.txt [hunk 1]
```diff
@@ -81,6 +81,7 @@ set(SRC_DIRS_LIST
                       "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/wake-on-lan-server"                      
                       "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/pump-configuration-and-control-server"
                       "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/power-source-configuration-server"
+                      "${CMAKE_SOURCE_DIR}/third_party/connectedhomeip/src/app/clusters/ota-requestor"
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/main/CMakeLists.txt:84]: @tima-q The support is now being reverted in #13437

Comment (tima-q) [reply examples/all-clusters-app/esp32/main/CMakeLists.txt:84]: Indeed, but if I add the OTA requestor it in the normal cluster build flow - and remove the stub, it becomes a dependency thing. It's not tied into ESP32 alone, but for any build using the requestor.
OTA Requestor needs an OTA Provider client, so without the combination, the app doesn't build anymore.
I see 1 option currently, untill ZAP is able to support both client/server roles, that's making the all-clusters-app an OTA provider client + OTA requestor server only.
Any thoughts on this ?

Comment (shubhamdp) [reply examples/all-clusters-app/esp32/main/CMakeLists.txt:84]: @tima-q, Please take a look here: https://github.com/project-chip/connectedhomeip/pull/13437#issuecomment-1009993943

Comment (tima-q) [reply examples/all-clusters-app/esp32/main/CMakeLists.txt:84]: @shubhamdp Thanks for the pointer to @bzbarsky-apple 's comment.
I can remove the server here if you like to keep it contained in 1 PR

Comment (shubhamdp) [reply examples/all-clusters-app/esp32/main/CMakeLists.txt:84]: I raised a PR #13454

Comment (tima-q) [reply examples/all-clusters-app/esp32/main/CMakeLists.txt:84]: ok - thanks - will match the changes here

**Comment by bzbarsky-apple** [comment]
@tima-q Please rebase on top of #13454 to make it clear which changes here are really new and which were already done in that PR?

**Comment by tima-q** [comment]
> @tima-q Please rebase on top of #13454 to make it clear which changes here are really new and which were already done in that PR?

Rebased - only changes are now in the data model build files + stub removal at application level. Thanks for following up.

**Review by shubhamdp — APPROVED** [review]

**Review by dhrishi — APPROVED** [review]

**Pull request merged**